store/helper/helper.go

```diff
@@ -672,6 +672,22 @@ func newIndexWithKeyRange(db *model.DBInfo, table *model.TableInfo, index *model
 	}
 }
 
+func newPartitionLocalIndexWithKeyRange(db *model.DBInfo, table *model.TableInfo, index *model.IndexInfo, partitionID int64) TableInfoWithKeyRange {
+	sk, ek := tablecodec.GetTableIndexKeyRange(partitionID, index.ID)
+	startKey := bytesKeyToHex(codec.EncodeBytes(nil, sk))
+	endKey := bytesKeyToHex(codec.EncodeBytes(nil, ek))
+	return TableInfoWithKeyRange{
+		&TableInfo{
+			DB:      db,
+			Table:   table,
+			IsIndex: true,
+			Index:   index,
+		},
+		startKey,
+		endKey,
+	}
+}
+
```

Comment:
how about modify `newIndexWithKeyRange` function and remove `newPartitionLocalIndexWithKeyRange`:

```shell
git diff
```
```go
diff --git a/store/helper/helper.go b/store/helper/helper.go
index f24443433e..ee811ecbdb 100644
--- a/store/helper/helper.go
+++ b/store/helper/helper.go
@@ -653,11 +653,11 @@ func newTableWithKeyRange(db *model.DBInfo, table *model.TableInfo) TableInfoWit

 // NewIndexWithKeyRange constructs TableInfoWithKeyRange for given index, it is exported only for test.
 func NewIndexWithKeyRange(db *model.DBInfo, table *model.TableInfo, index *model.IndexInfo) TableInfoWithKeyRange {
-       return newIndexWithKeyRange(db, table, index)
+       return newIndexWithKeyRange(db, table, index, table.ID)
 }

-func newIndexWithKeyRange(db *model.DBInfo, table *model.TableInfo, index *model.IndexInfo) TableInfoWithKeyRange {
-       sk, ek := tablecodec.GetTableIndexKeyRange(table.ID, index.ID)
+func newIndexWithKeyRange(db *model.DBInfo, table *model.TableInfo, index *model.IndexInfo, physicalID int64) TableInfoWithKeyRange {
+       sk, ek := tablecodec.GetTableIndexKeyRange(physicalID, index.ID)
```

Reply:
Good advice.